packages/devtools_app/lib/src/service/editor/editor_client.dart

```diff
@@ -57,3 +78,208 @@ abstract class EditorClient {
   /// the appropriate updates).
   Future<void> enablePlatformType(String platformType);
 }
+
+/// An implementation of [EditorClient] that connects to an editor over DTD.
+///
+/// Changes made to the editor services/events should be considered carefully to
+/// ensure they are not breaking changes to already-shipped editors.
+class DtdEditorClient extends EditorClient {
+  // TODO(dantup): Merge this into EditorClient once the postMessage version
+  //  is removed.
+
+  DtdEditorClient(this._dtd) {
+    unawaited(initialized); // Trigger async initialization.
```

Comment:
We can't remove the variable because it's part of the external interface (the sidebar will fetch the initial set of devices/debug sessions once this completes).